### PR TITLE
Scio 0.8.0 Migration

### LINKED ISSUE #204: Scio 0.8.0 Migration
Ugprade Scio and it's dependencies for 0.8.0 compatibility. Some minor overlap with #176

### PR DESCRIPTION
**Pull request by idreeskhan**
Currently built on beta2. Fixes #204

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,29 +20,29 @@ import Keys._
...
 val avroVersion = "1.8.2"
-val gcsVersion = "hadoop2-1.9.17"
+val beamVersion = "2.16.0"
+val bigqueryVersion = "v2-rev20181104-1.27.0"
+val gcsVersion = "hadoop2-2.0.0"
+val guavaVersion = "25.1-jre"
 val hadoopVersion = "2.7.7"
-val jodaTimeVersion = "2.9.9"
+val jodaTimeVersion = "2.10.5"
+val magnoliaVersion = "0.12.2" // needs to stay in sync with scio version
```
Comment (idreeskhan) [inline build.sbt]: This will conflict with #203

File: build.sbt [hunk 2]
```diff
@@ -20,29 +20,29 @@ import Keys._
 
 val algebirdVersion = "0.13.6"
 val avroVersion = "1.8.2"
-val gcsVersion = "hadoop2-1.9.17"
+val beamVersion = "2.16.0"
```
Comment (anne-decusatis) [inline build.sbt]: scio 0.8.0 uses beam 2.17.0

File: build.sbt (hunk 1 above)
Comment (anne-decusatis) [reply build.sbt]: 0.12.6 on 0.8.0 release

File: build.sbt [hunk 3]
```diff
@@ -20,29 +20,29 @@ import Keys._
...
 val hadoopVersion = "2.7.7"
 val jodaTimeVersion = "2.10.5"
+val magnoliaVersion = "0.12.2" // needs to stay in sync with scio version
 val parquetVersion = "1.10.1"
 val protoBufVersion = "3.11.1"
+val scalaTestVersion = "3.1.0"
 val scalaCheckVersion = "1.14.3"
-val scalaTestVersion = "3.0.8"
-val scioVersion = "0.7.4"
+val scioVersion = "0.8.0-beta2"
```
Comment (anne-decusatis) [inline build.sbt]: update to 0.8.0 non-beta

**Comment by anne-decusatis** [comment]
I also think the failing tests are due to the scalatest upgrade to 3.1.0 which is in scio 0.8.0 release, but not in 0.8.0-beta2. Going to try upgrading that first before investigating further

**Pull request merged**